modules/swagger-codegen/src/main/resources/php/api.mustache

```diff
@@ -122,7 +122,36 @@ use \{{invokerPackage}}\ObjectSerializer;
         // verify the required parameter '{{paramName}}' is set
         if (${{paramName}} === null) {
             throw new \InvalidArgumentException('Missing the required parameter ${{paramName}} when calling {{operationId}}');
-        }{{/required}}{{/allParams}}
+        }
+        {{#hasValidation}}
+        {{#maxLength}}
+        if (strlen(${{paramName}}) > {{maxLength}}) {
+            throw new \InvalidArgumentException('invalid length for "${{paramName}}" when calling {{classname}}.{{operationId}}, must be smaller than or equal to {{maxLength}}.');
+        }
+        {{/maxLength}}
+        {{#minLength}}
+        if (strlen(${{paramName}}) > {{minLength}}) {
+            throw new \InvalidArgumentException('invalid length for "${{paramName}}" when calling {{classname}}.{{operationId}}, must be bigger than or equal to {{minLength}}.');
+        }
+        {{/minLength}}
+        {{#maximum}}
+        if (${{paramName}} > {{maximum}}) {
+            throw new \InvalidArgumentException('invalid value for "${{paramName}}" when calling {{classname}}.{{operationId}}, must be smaller than or equal to {{maximum}}.');
+        }
+        {{/maximum}}
+        {{#minimum}}
+        if (${{paramName}} < {{minimum}}) {
+            throw new \InvalidArgumentException('invalid value for "${{paramName}}" when calling {{classname}}.{{operationId}}, must be bigger than or equal to {{minimum}}.');
+        }
+        {{/minimum}}
+        {{#pattern}}
+        if (!preg_match("{{pattern}}", ${{paramName}})) {
+            throw new \InvalidArgumentException('invalid value for "{{paramName}}" when calling {{classname}}.{{operationId}}, must conform to the pattern {{pattern}}.');
+        }
+        {{/pattern}}
+
+        {{/hasValidation}}
+        {{/required}}{{/allParams}}
```

Comment:
I would suggest moving {{/required}} to a new line between L125 and L126 to cover all parameters (required and optional)